api/handlers.go

```
@@ -1124,3 +1121,16 @@ func notImplementedHandler(c *context, w http.ResponseWriter, r *http.Request) {
 func optionsHandler(c *context, w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
+
+// Decode a base64-url-safe-encoded Registry Auth Config header.
+func decodeRegistryAuthHeader(r *http.Request, header string) []byte {
+	auth := r.Header.Get(header)
+	if auth == "" {
+		return nil
+	}
+	buf, err := base64.URLEncoding.DecodeString(auth)
+	if err != nil {
+		return nil
```

I do not like silently ignoring errors.